### ISSUE TITLE
Service Locations not respecting Appointments "Use main facility phone number"

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/va.gov-cms#17511]
## Describe the defect
With the service location migration, if "Use the facility phone number" is selected under "Appointments", this is not being respected and the phone number is not being shown on the front end. 

## To Reproduce
- [Rheumatology service in the Santa Maria VA Clinic](https://web-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/greater-los-angeles-health-care/locations/santa-maria-va-clinic/#rheumatology)
  - This currently shows no phone numbers in the integration branch
  - It DOES however show a main number in [prod](https://www.va.gov/greater-los-angeles-health-care/locations/santa-maria-va-clinic/#rheumatology)
- In [the CMS for this service](https://pr15622-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/node/64056/edit), under the Appointments section, "Use the facility phone number" is checked
  - No other phone numbers are added to the Appointments section not to the Contact Info section

## AC / Expected behavior

- [ ] Facility phone number should show on FE when "Use the facility phone number" under appointments section is checked

### Team
Please check the team(s) that will do this work.

- [ ] `CMS Team`
- [ ] `Public Websites`
- [x] `Facilities`
- [ ] `User support`
- [ ] `Accelerated Publishing`

**Comment by eselkin** [comment]
So the problem is: VBA facilities have a `fieldPhoneNumber` but regular VAMC do not (they have `facility.fieldPhoneNumber`). Non-clinical services had added a `with` property to the drupal template to cover this condition but regular VAMC services did not. I added it and will add it to the content-build branch we're on in testing.

**Comment by eselkin** [comment]
See service location integration site for testing. Changes made on that branch.

**Comment by jilladams** [comment]
Merged into Service Location branch.

**Comment by davidmpickett** [comment]
It's possible that there was an overcorrection here. I am now seeing instances where the Main Phone number for a Facility is showing up even when both instances of use the facility phone number are unchecked.

- [Nephrology at pittsburgh - FE](https://web-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/pittsburgh-health-care/locations/pittsburgh-va-medical-center-university-drive/#nephrology)
- [Nephrology at pittsburgh - Drupal](https://pr15622-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/pittsburgh-health-care/locations/pittsburgh-va-medical-center-university-drive/nephrology)

 
![Screenshot 2024-03-22 at 2 21 45 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/51967950/832a26be-cd38-4846-9bbf-4760322a18ce)

![Screenshot 2024-03-22 at 2 22 06 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/51967950/7e8f3b53-ac14-48f5-8498-b30fbcc3ec14)

![Screenshot 2024-03-22 at 2 22 17 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/51967950/a71a04bb-6ef5-40e6-9c31-c60729b0621f)

**Comment by davidmpickett** [comment]
@eselkin Re my comment above this. I think that issue might be at [line 131 of the Service Location template](https://github.com/department-of-veterans-affairs/content-build/blob/8812728f2b7a395fc6c6b45c1c1d275897147a02/src/site/paragraphs/service_location.drupal.liquid#L131)

I think it's missing a check for the check box.
`{% if single.fieldUseMainFacilityPhone and serviceLocationMainNumber %}`

**Comment by eselkin** [comment]
Yeah, seems like that's the issue. But do we only include the main facility number in that condition?

**Comment by jilladams** [comment]
Confirming yes: checkboxes are Appointments accordion > Phone number for appointments, and Contact info > Phone in Drupal. 

I'm gonna move this back to In Progress, since we have a follow up action that needs to be fixed / merged to the Integration branch before we can say we're "complete pending integration."

**Comment by davidmpickett** [comment]
@eselkin Are you waiting for additional review/QA on your PR? I see @maxx1128 approved last week. Is your PR ready to merge into the Content Build Integration Branch?

**Comment by eselkin** [comment]
It should be. I had to update something from main for a test to pass

**Comment by davidmpickett** [comment]
Testing on Tugboat revealed that the logic needs to be:
- The use facility phone check box only controls whether the facility number shows or not 
- Additional numbers can be added regardless of that checkbox state and should also show

**Comment by jilladams** [comment]
PR merged to integration branch.

**Comment by davidmpickett** [comment]
Confirmed on Tugboat 

https://web-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/pittsburgh-health-care/locations/pittsburgh-va-medical-center-university-drive/#nephrology

![Screenshot 2024-04-18 at 5 46 04 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/51967950/00da0add-6e72-484c-afd9-4d5a04b726a0)

https://web-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/greater-los-angeles-health-care/locations/santa-maria-va-clinic/#rheumatology

![Screenshot 2024-04-18 at 5 46 22 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/51967950/6840ba26-7a4a-4fc5-b75a-7905d01b8264)
